tests/functional/event_handler/test_api_middlewares.py

```diff
@@ -397,6 +397,46 @@ def dummy_route():
     assert result["statusCode"] == 200
 
 
+@pytest.mark.parametrize(
+    "app, event",
+    [
+        (ApiGatewayResolver(proxy_type=ProxyEventType.APIGatewayProxyEvent), API_REST_EVENT),
+        (APIGatewayRestResolver(), API_REST_EVENT),
+        (APIGatewayHttpResolver(), API_RESTV2_EVENT),
+    ],
+)
+def test_api_gateway_middleware_with_include_router_prefix(app: EventHandlerInstance, event):
+    # GIVEN a router instance
+    router = Router()
+
+    def app_middleware(app: EventHandlerInstance, next_middleware: NextMiddleware):
+        # inject a variable into resolver context
+        app.append_context(injected="injected_value")
+        response = next_middleware(app)
+
+        return response
+
+    # WHEN we register a route with a middleware
+    @router.get("/path", middlewares=[app_middleware])
+    def dummy_route():
+        # if the middleware is executed, the context variable will be available
+        # Otherwise, an exception will be raised
+        injected_context = app.context["injected"]
+
+        assert injected_context == "injected_value"
+
+        return Response(status_code=200, body="works!")
+
+    # WHEN register the route with a prefix
+    app.include_router(router, prefix="/my")
+
+    # THEN resolving a request must execute the middleware
+    # THEN return a successful response http 200 status code
```

Comment:
```suggestion
    # THEN resolving a request must execute the middleware
    # and return a successful response http 200 status code
```

suggestion(non-blocking): makes it easier to parse that we're asserting two behaviours only: 1/ Route runs successfully (non-404), and 2/ Middleware runs when a prefix is in place